Swap post processing address issue

gemdev111 opened the pull request:
Updates transaction post processing logic to support new TransactionWallet with correct from,to address
Fixes swap to value address not correct in cross-chain swaps
closes #1008 
closes #1004

gemdev111 committed:
Introduce TransactionWallet and refactor transaction handling

Added the TransactionWallet type to encapsulate a transaction and its wallet, along with related extensions and test coverage.
 Refactored transaction processing, state updates, and service interfaces to use TransactionWallet instead of separate transaction and wallet identifiers.
Updated database access and post-processing logic to support the new structure, improving clarity and maintainability.

gemdev111 committed:
Make TransactionWallet conform to Codable and Equatable

Added Codable and Equatable conformances to the TransactionWallet struct to enable encoding, decoding, and value comparison. Also updated the file header to indicate it was generated by typeshare.

gemdev111 committed:
Improve swap transaction metadata handling and test mock

Added logging when swap transaction metadata is missing in WalletTransaction+Primitives.swift for better debugging. Updated Transaction+PrimitivesTestKit.swift to use the provided assetId in mock transactions instead of always using a mock assetId.

gemdev111 committed:
Fix spacing in WalletTransactionTests and rename service file

Corrected spacing in WalletTransactiontTests.swift for better code style consistency. Renamed TransactionPostProcessingService .swift to TransactionPostProcessingService.swift to remove the extra space in the filename.

gemcoder21 commented on Packages/Primitives/Sources/Extensions/WalletTransaction+Primitives.swift:
```diff
@@ -0,0 +1,43 @@
+        switch transaction.type {
+        case .swap:
+            guard case let .swap(meta) = transaction.metadata else {
+                NSLog("Metadat not available for swap transaction:\(transaction) in wallet: \(wallet.id)")
```
```suggestion
                NSLog("Metadata not available for swap transaction:\(transaction) in wallet: \(wallet.id)")
```

gemdev111 committed:
Refactor balance update logic and remove unused address method

Removed the address(for:) method from TransactionWallet and its associated tests. Updated the BalancerUpdater protocol to support batch balance updates for a wallet and asset IDs. Refactored TransactionStateUpdatePostJob to use the new batch update method, simplifying the balance update process after transactions.

gemdev111 committed:
Merge branch 'main' into 990-swap-address-issue

gemdev111 committed:
Refactor transaction post-processing to use wallet and transaction

Removed WalletTransaction+Primitives extension and updated TransactionPostProcessingService to accept Wallet and Transaction separately instead of TransactionWallet. Adjusted related method calls and updated TransactionStateUpdateJob to match the new interface. This refactor simplifies the data flow and removes unnecessary extension logic.

gemdev111 committed:
Merge branch '990-swap-address-issue' of https://github.com/gemwalletcom/gem-ios into 990-swap-address-issue

gemdev111 committed:
Update balance updater to include fee asset ID

The balance updater now processes both transaction asset IDs and the fee asset ID, ensuring all relevant balances are updated after a transaction.
